libdevcore/MemoryDB.cpp

```diff
@@ -100,33 +96,18 @@ void MemoryDB::insert(h256 const& _h, bytesConstRef _v)
 #endif
 }
 
-bool MemoryDB::kill(h256 const& _h)
+void MemoryDB::kill(h256 const& _h)
 {
 #if DEV_GUARDED_DB
 	ReadGuard l(x_this);
 #endif
+	if (_h == EmptyTrie)
```

Comment:
To me it looks like this check should be done in calling code instead. MemoryDB should not care about how trie is implemented